### PR TITLE
Convert server to null safety.

### PR DESCRIPTION
**Pull request by polina-c**

### CODE REVIEW

File: packages/devtools_app/lib/src/config_specific/server/_server_web.dart [hunk 1]
```diff
@@ -75,10 +72,10 @@ Future<bool> setAnalyticsEnabled([bool value = true]) async {
       '?$devToolsEnabledPropertyName=$value',
     );
     if (resp?.status == HttpStatus.ok) {
-      assert(json.decode(resp.responseText) == value);
+      assert(json.decode(resp!.responseText!) == value);
       return true;
     } else {
-      logWarning(resp, apiSetDevToolsEnabled, resp.responseText);
+      logWarning(resp, apiSetDevToolsEnabled, resp!.responseText);
```
Comment (kenzieschmoll) [inline packages/devtools_app/lib/src/config_specific/server/_server_web.dart:78]: use ? here instead. resp may be null

File: packages/devtools_app/lib/src/config_specific/server/_server_web.dart [hunk 2]
```diff
@@ -278,40 +275,40 @@ Future<void> setLastShownReleaseNotesVersion(String version) async {
 Future<void> resetDevToolsFile() async {
   if (isDevToolsServerAvailable) {
     final resp = await _request(apiResetDevTools);
-    if (resp?.status == HttpStatus.ok) {
-      assert(json.decode(resp.responseText));
+    if (resp!.status == HttpStatus.ok) {
```
Comment (kenzieschmoll) [inline packages/devtools_app/lib/src/config_specific/server/_server_web.dart:278]: use ? instead. resp may be null

File: packages/devtools_app/lib/src/shared/survey.dart [hunk 3]
```diff
@@ -40,8 +40,8 @@ class SurveyService {
     if (!server.isDevToolsServerAvailable) return null;
 
     _cachedSurvey ??= await _fetchSurveyContent();
-    if (_cachedSurvey != null) {
-      await server.setActiveSurvey(_cachedSurvey!.id);
+    if (_cachedSurvey != null && _cachedSurvey!.id != null) {
```
Comment (kenzieschmoll) [inline packages/devtools_app/lib/src/shared/survey.dart:43]: why not `if (_cachedSurvey?.id != null)` instead?

File: packages/devtools_app/lib/src/config_specific/server/_server_web.dart (hunk 1 above)
Comment (polina-c) [reply packages/devtools_app/lib/src/config_specific/server/_server_web.dart:78]: done

File: packages/devtools_app/lib/src/config_specific/server/_server_web.dart (hunk 2 above)
Comment (polina-c) [reply packages/devtools_app/lib/src/config_specific/server/_server_web.dart:278]: done

File: packages/devtools_app/lib/src/shared/survey.dart (hunk 3 above)
Comment (polina-c) [reply packages/devtools_app/lib/src/shared/survey.dart]: done

**Review by kenzieschmoll — APPROVED** [review]

**Comment by polina-c** [comment]
tested

**Pull request merged**